### PR TITLE
fix: exclude non-running pods from podlist

### PR DESCRIPTION
**Pull request by huettner94**
## Summary

previously the executor would happily set `TARGET_PODS` for the backup job to the pod of a previous running backup instead of a actually running pod.
This prevents `k8up.io/backupcommand` from working, as then the backup job would actually check the old incarnation of itself instead of the real target pod.

### For Code changes

- [x] Categorize the PR by setting a good title and adding one of the labels:
  `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
  as they show up in the changelog
- [x] PR contains the label `area:operator`
- [X] Link this PR to related issues
- [X] I have not made _any_ changes in the `charts/` directory.

### CODE REVIEW

**Comment by huettner94** [comment]
Unfortunately it does not seem like i can set labels on this MR

**Comment by huettner94** [comment]
Thanks @Kidswiss for the feedback.

During fixing the issue i found that we already set a label on all pods used for backups. I then decided that using this label would be more reliable than filtering for running pods.

(also a integrationtest is now added)

**Comment by Kidswiss** [comment]
> Thanks @Kidswiss for the feedback.
> 
> During fixing the issue i found that we already set a label on all pods used for backups. I then decided that using this label would be more reliable than filtering for running pods.
> 
> (also a integrationtest is now added)

Yeah, you're right, that makes more sense. I've triggered the tests, let's see how they go.

File: operator/backupcontroller/executor.go
Comment (Kidswiss) [inline operator/backupcontroller/executor.go:65]: The lint job is complaining here to not use an unkeyed field in the `client.MatchingLabelsSelector` literal.

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Just one small linting issue and then it should be ready for merging.

File: operator/backupcontroller/executor.go
Comment (huettner94) [reply operator/backupcontroller/executor.go]: fixed

**Comment by huettner94** [comment]
thanks fixed

**Comment by Kidswiss** [comment]
@huettner94 unfortunately the linter found something new to complain 😅 

You can run the linter locally by using `make lint`. Then you won't have to wait for me to trigger the CI/CD every time.

**Comment by huettner94** [comment]
Just pushed a fix for the lint. I could not reproduce the issue with the e2e tests locally. they worked fine for me

**Comment by Kidswiss** [comment]
Hi @huettner94 

Sorry for the late answer was on a vacation.

The e2e looks good now. Sometime they can be a bit fiddly....

Can you please run `make docs-all` and then commit the changes? That should shut up the linter for good.

After that I think we're ready to merge here.

**Comment by huettner94** [comment]
> Hi @huettner94
> 
> Sorry for the late answer, was on a vacation.
> 
> The e2e looks good now. Sometime they can be a bit fiddly....
> 
> Can you please run `make docs-all` and then commit the changes? That should shut up the linter for good.
> 
> After that I think we're ready to merge here.

No problem, vacation is always great.

There are some issues with the `make docs-all`:
1. It references some `playbook.yml` in `$(out_dir)/index.html: playbook.yml $(pages)`, but this file does no longer exist. It was removed in https://github.com/k8up-io/k8up/pull/446
2. The used docker image `vshn/antora:2.3.3` seems to no longer exist on dockerhub (same goes for `vshn/asciidoctor-pdf:1.8.1`)

If i take the `playbook.yml` from the old commit i get to the needed docker images, but there i cant really fix something

**Comment by Kidswiss** [comment]
Huh, weird that the image is no longer there. I feel like this needs more investigation with our docs guy and is a separate issue...

I'm going ahead and merge it anyway and fix the linting/docs in another PR.

Thanks for the contribution! 🎉

**Pull request merged**
